docs/blueprints/Xen_Late_Launch.md

```diff
@@ -0,0 +1,51 @@
+Xen Late Launch Hypervisor
+==========================
+
+## Summary
+
+Xen hypervisor was extended to make it function as the late launch hypervisor
+for both AMD SKINIT and Intel TXT.  The changes have not yet been upstreamed and
+currently target packages for Qubes OS.
+
+## Background
+
+The late launch process provides means to measure a target execution
+environment as prescribed by the CPU's late launch protocol before jumping into
+the environment. Thereby establishing a hardware-based Root of Trust for
+Measurement (RTM).
+
+## Approach
+
+File format changes:
+
+ * Added an MLE header (defined for Intel TXT but usable on AMD where DCE is
+   controlled by us) to provide basic information to the bootloader including
+   an entry point
+ * Added a new entry point specific for the late launch to know how Xen was
+   started and be able to change behaviour based on that knowledge
+
+Control flow changes:
+
+ 1. Start Xen from GRUB2 via Multiboot2 protocol but through the entry point
+    from MLE header
+ 2. Figure out location of SLRT and MBI:
+    - [TXT] Extract addresses from OS2MLE structure
+    - [SKINIT] Addresses are available in `%EBP` and `%EBX` registers
+      respectively
+ 3. Verify that PMR ranges are sane and provide protections for all involved
```

Comment:
Note that this is TXT only.

Reply:
https://github.com/TrenchBoot/TrenchBoot.github.io/compare/97941c079ed7b202e52b9bc265febac7b8afd04f..1a18d56ac6cad3dcb05e6dddf5ba68cdc918abfd#diff-ca60d9a35b5aec758ba42ecc5247bafcae97227e45682dcc2af1bf4afadf84df